src/rapids_dependency_file_generator/schema.json

```diff
@@ -0,0 +1,160 @@
+                        }
+                    },
+                    "minProperties": 1,
+                    "additionalProperties": false
+                }
+            }
+        },
+        "channels": {"$ref": "#/$defs/Channels"}
+    },
+    "required": ["files", "dependencies"],
+    "additionalProperties": false,
+    "$defs": {
+        "Channel": {
+            "type": "string",
+            "format": "iri-reference"
+        },
+        "ChannelList": {
+            "type": "array",
+            "items": {
+                "$ref": "#/$defs/Channel"
+            }
+        },
+        "Channels": {
+            "$oneOf": [
+                {"$ref": "#/$defs/Channel"},
+                {"$ref": "#/$defs/ChannelList"}
+            ]
+        },
+        "Matrix": {
+            "type": "object",
+            "patternProperties": {
+                ".*": {
+                    "type": "array",
+                    "items": {"type": "string"}
+                }
+            }
+        },
+        "MatrixMatcher": {
+            "type": "object",
+            "properties": {
+                "matrix": {
+                    "oneOf": [
+                        {
+                            "type": "object",
+                            "patternProperties": {
+                                ".*": {"type": "string"}
+                            }
+                        },
+                        {"type": "null"}
+                    ]
+                },
+                "packages": {"oneOf": [
+                    {"$ref": "#/$defs/Requirements"},
+                    {"type": "null"}
+                ]}
+            },
+            "requiredProperties": ["matrix", "packages"],
+            "additionalProperties": false
+        },
+        "Matrices": {
+            "type": "array",
+            "items": {"$ref": "#/$defs/MatrixMatcher"}
+        },
+        "OutputTypes": {
+            "enum": ["conda", "requirements"]
+        },
+        "OutputList": {
+            "type": "array",
+            "item": {"$ref": "#/$defs/OutputTypes"}
+        },
+        "Outputs": {
+            "oneOf": [
+                {"$ref": "#/$defs/OutputTypes"},
+                {"$ref": "#/$defs/OutputList"},
+                {"const": "none"}
+            ]
+        },
+        "Packages": {
+            "type": "array",
+            "items": {
+                "oneOf": [
+                    {"$ref": "#/$defs/Requirement"},
+                    {"$ref": "#/$defs/PipRequirements"}
+                ]
+            }
+        },
+        "Requirement": {
+            "type": "string"
+        },
+        "Requirements": {
+            "type": "array",
+            "items": {
+                    "$ref": "#/$defs/Requirement"
+                },
+            "minItems": 1
+
+        },
+        "PipRequirements": {
+            "type": "object",
+            "properties": {
+                "pip": {"$ref": "#/$defs/Requirements"}
```

Comment:
Should pip support oneOf Requirement or Requirements? I'm actually not 100% sure what the syntax is for conda environments with only one pip requirement and whether that can be on a single line (not that it exactly matters here, more just thinking about how consistent dependencies.yaml would look when compared to a conda env.yaml for readability), I'm mostly asking for the sake of consistency with Packages since PipRequirements is essentially a sublist of packages.

Reply:
All examples for the `dependencies.yaml` file I've seen have used a list. We will probably need a bunch of `list_or_string()` functions within the parse so I don't think it's too much of an implementation headache to support both. What would you prefer?

Reply:
I have also never seen such a dependencies.yaml file. I don't even think I have seen a conda environment.yaml file that doesn't use a list for pip dependencies. However, I just tried it locally for a conda env and it does work. If environment.yaml files support it I think dependencies.yaml should too, and if we're writing a schema we may as well be precise, so I say yes let's support both here.